cmd/activator/main.go

```diff
@@ -158,23 +157,24 @@ func main() {
 		logger.Fatalw("Failed to construct network config", zap.Error(err))
 	}
 
-	// Enable TLS against queue-proxy when internal-encryption is enabled.
-	tlsEnabled := networkConfig.InternalEncryption
-
+	// Enable TLS both as client and as server if DataplaneTrust != TrustDisabled
+	tlsEnabled := networkConfig.DataplaneTrust != netcfg.TrustDisabled
+	var roundtrip http.RoundTripper
 	var certCache *certificate.CertCache
 
-	// Enable TLS client when queue-proxy-ca is specified.
 	// At this moment activator with TLS does not disable HTTP.
 	// See also https://github.com/knative/serving/issues/12808.
 	if tlsEnabled {
-		logger.Info("Internal Encryption is enabled")
-		certCache = certificate.NewCertCache(ctx)
-		transport = pkgnet.NewProxyAutoTLSTransport(env.MaxIdleProxyConns, env.MaxIdleProxyConnsPerHost, &certCache.TLSConf)
+		logger.Info("Dataplane trust %q is used", networkConfig.DataplaneTrust)
+		certCache = certificate.NewCertCache(ctx, networkConfig.DataplaneTrust)
+		roundtrip = activatorhandler.NewProxyAutoTLSTransport(env.MaxIdleProxyConns, env.MaxIdleProxyConnsPerHost, &certCache.ClientTLSConf)
+	} else {
+		roundtrip = activatorhandler.NewProxyAutoTransport(env.MaxIdleProxyConns, env.MaxIdleProxyConnsPerHost)
```

Comment:
I changed back to transport.
We should not drop the else clause. We no longer create a non-tls transport when we are configured to work with tls (since it makes no sense). hence we need the else clause.